sql/core/src/test/scala/org/apache/spark/sql/DataFrameStatSuite.scala

```diff
@@ -126,6 +126,19 @@ class DataFrameStatSuite extends QueryTest with SharedSparkSession {
     assert(math.abs(corr3 - 0.95723391394758572) < 1e-12)
   }
 
+  test("SPARK-30532 pearson correlation to understand fully-qualified column name") {
```

Comment:
can we merge these tests into one? Then we can share the below code
```
val df1 = spark.sparkContext.parallelize(0 to 10).toDF("num").as("table1")
val df2 = spark.sparkContext.parallelize(0 to 10).toDF("num").as("table2")
val dfx = df2.crossJoin(df1)
```

Reply:
Sure, I will submit update shortly.

Reply:
Done!